buffer_pool.go

```diff
@@ -9,18 +9,20 @@ import (
 var bufferPool sync.Pool
 
 func getPacketBuffer() []byte {
```

Comment:
Can you run a benchmark with this changed to return *[]byte? That way you wouldn't need to allocate another copy of the empty slice before reslicing it.